### ISSUE TITLE
Size conversion error

### ISSUE THREAD

**Issue by hogren** [wkhtmltopdf/wkhtmltopdf#4107]
Hello,
I found a problem of size (for height and width CSS properties) conversion.

I give a minimal code to reproduce the problem : 

```html
<html>
    <head>
        <style> 
        * {
            margin: 0 0 0 0;
        }

        body {
            background-color: orange;
        }

        .tiers1 {
            height: 99mm;
            width: 100%;
            background-color: blue;
        }

        .tiers2 {
            height: 33.333%;
            width: 100%;
            background-color: yellow;
        }

        .tiers3 {
            height: 33.333%;
            width: 100%;
            background-color: green;
        } 
        </style>
    </head>
    <body>
        <div class="tiers1"></div>
        <div class="tiers2"></div>
        <div class="tiers3"></div>
    </body>
</html>

```

If I make `33.333%` for the height of the three DIV, it works. The problem is if I try to enter a "length" value (mm, cm, pt, etc.). As you can see if you run this command : 
`wkhtmltopdf -B 0 -L 0 -R 0 -T 0 -s A4  file.html file.pdf`
The DIV with a "length" value, is smaller than it has to be.

I investigated the pdf file. I deflated the content. I saw that `33.333%` for the height is converted to `467 points`, and `99mm (297 (A4 height) /3)` is converted to `374` points.

I tested with wkhtmltopdf 0.12.5 (with patched qt) on a Debian 9 Docker, and wkhtmltopdf 0.12.4 (with patched qt) on Gentoo.

Thank you very much for your help !

Regards,

Hogren

**Comment by CPASimUSante** [comment]
I had some similar problems #4113  and found out that using the --disable-smart-shrinking option helped a lot. By default the --enable-smart-shrinking is activated. [https://wkhtmltopdf.org/usage/wkhtmltopdf.txt](url)
Hope it helps

**Comment by hogren** [comment]
Ok CPASimUSante !
I make a quick test. It seems to work. Thanks.

For the Dev team : Ok sorry it's not a bug of wkhtmltopdf. But : 
It will be very helpful I think, if "disable-smart-shrinking" option appear in Help and not only in Extended-Help.

Please note this query for other people like me who want a precise rendering, and close my issue when your decision is taken.

Thank you a lot !

And Thank you again CPASimUSante !

**Comment by Tomsgu** [comment]
Why do you think `--disable-smart-shrink` would be better to put it in the normal help and why not put others? Haven't you found the `-O -L -R -T` options also in the extended help? Note that in unpatched version there is no `--disable-smart-shrink`.

**Comment by hogren** [comment]
Tomsgu, you're right!
I just thought that it's not "normal" to no synchronize sizes of html objects by default (when we give precise sizes). I suppose this default comportment can't be altered (compatibility reasons). So, as it's not "normal", but a default behavior, users need to see it as a principal option when they look help.

It's your decision, I just propose. I made an error, I read --help but not --extended-help (-A -B -T, etc. were given to me by a colleague). I think that I am not the first one (sure, see CPASimUSante issue), and not the last one to make the error.

Thanks for your response.
Regards,
Hogren

**Comment by hogren** [comment]
@Tomsgu 
It's not that I did not want to read the extended-help. My first thinking is that it's not normal that when I say «the table must have a 9cm (or equivalent in pt)», the table measure 7cm so it's probably a bug.
I really think that there was a calculation error.
I am really frustrated because I often read documentations. I like read documentation before use a tool, to know what is possible. I made a mistake with this tool. 

I will try to think more about that and I will may be propose a pull request.
Thanks again @Tomsgu @CPASimUSante .

**Comment by Tomsgu** [comment]
@hogren yes I agree with your point. I didn't want to say you don't read a documentation, because as I pointed the `--disable-smart-shrinking` option should have a better description. That's the first main problem I think.

I am leaving this open though, until we have a good fix. Please feel free to open a PR @hogren.
